src/api/app/models/branch_package.rb

```diff
@@ -507,20 +498,13 @@ def find_packages_to_branch
       req = BsRequest.find_by_number(params[:request])
 
       req.bs_request_actions.each do |action|
-        prj=nil
-        pkg=nil
-        if action.source_project || action.source_package
-          if action.source_package
-            pkg = Package.get_by_project_and_name action.source_project, action.source_package
-          elsif action.source_project
-            prj = Project.get_by_name action.source_project
-          end
+        if action.source_package
+          pkg = Package.get_by_project_and_name action.source_project, action.source_package
         end
 
         @packages.push({ link_target_project: action.source_project, package: pkg, target_package: "#{pkg.name}.#{pkg.project.name}" })
       end
     elsif params[:project] && params[:package]
-      pkg = nil
```

Comment:
This can't be removed line 540 is using this local variable that can be not assigned at line 530.. so it can be undefined